corehq/apps/api/resources/v0_5.py

```diff
@@ -1376,3 +1378,21 @@ def get_datasource_data(request, config_id, domain):
     query = cursor_based_query_for_datasource(request_params, datasource_adapter)
     data = response_for_cursor_based_pagination(request, query, request_params, datasource_adapter)
     return JsonResponse(data)
+
+
+@require_GET
+@api_auth()
+@requires_privilege_with_fallback(privileges.API_ACCESS)
+@api_throttle
+def get_cca_user_roles(request, domain):
```

Comment:
Hey @Charl1996 

Can we consider using a class based view and inherit from the ones already defined?

It seems in HQ APIs we have always had class based views with `get_ucr_data` being the first method based view.
Both ways are okay though by using the structure that is being used in class based views in HQ APIs, and inheriting from predefined base view classes, keeps us aligned us the approach being followed in other API end points. For example, keeping a track of the decorators to be added on this view is going to be hard and risky.

If you wish to push this to be more generic/GA in future,
You could consider adding a UserRoleResource though that is not a requirement for this ticket.

Reply:
Mmm...I get your sentiment, but changing this to a class based view feels like a lot of code to do something this simple. 

That said, I like the idea of having a more generic UserRoleResource, so here's my suggestion: what if we create the UserRoleResource (as part of a separate PR) and then refactor this view into that more generic view?

Reply:
> Mmm...I get your sentiment, but changing this to a class based view feels like a lot of code to do something this simple.

I am not sure why we chose to do APIs as class based views when this was introduced. One thing comes to mind is that its easier to extend functionality using subclasses. I guess same thing would be hard to maintain in view functions.

I assume the rewrite to a class based view should be straight forward. Let me know if it's otherwise and I would be happy to assist. I believe if not now, but some day later someone would need to convert this to a class based view for code maintainability.
If its blocking feel free to add a ticket to do this refactor as a follow up instead.

Reply:
Cool. I'm currently checking if I can't simply pass the roles in the oauth token as part of the scope, so maybe this isn't necessary. Will update when I know more.

Reply:
+1 to the plan to use a class-based view. It might be possible to make this a method of WebUserResource (line 376), since it is only applicable to WebUsers.

Reply:
See [this commit](https://github.com/dimagi/commcare-hq/pull/34987/commits/83cafc89defb11fb8d52a5192d56e0ae52ee90be).